### PR TITLE
Homework oop1

### PR DESCRIPTION
**Pull request by malinaq3**
Authors: @wbrejecki @McKucia @Ian64 @kali-03 @DawidMalarz @malinaq3

### CODE REVIEW

File: shm/Map.cpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#include "Map.hpp"
+#include <algorithm>
+#include <iostream>
+#include <random>
+#include <set>
```
Comment (WBigaj) [inline shm/Map.cpp:5]: I think there should be new line between header include and standard library includes

File: shm/Map.hpp [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include <vector>
+#include "Island.hpp"
```
Comment (WBigaj) [inline shm/Map.hpp:4]: same here.

File: shm/Player.hpp [hunk 3]
```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include "Ship.hpp"
+#include "Cargo.hpp"
```
Comment (WBigaj) [inline shm/Player.hpp:4]: Includes should be sorted alphabetically.

**Review by WBigaj — COMMENTED** [review]
I have checked your code and given my feedback.

File: shm/Cargo.hpp [hunk 4]
```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <string>
+
+class Cargo {
+public:
+    Cargo(std::string n = "", size_t a = 0, size_t bp = 0):
+        name_(n), amount_(a), basePrice_(bp) {};
```
Comment (nauka-programowania-MA) [inline shm/Cargo.hpp:8]: move to .cpp

File: shm/Coordinates.hpp [hunk 5]
```diff
@@ -0,0 +1,12 @@
+#pragma once
+
+class Coordinates {
+public:
+    Coordinates() = default;
```
Comment (nauka-programowania-MA) [inline shm/Coordinates.hpp:5]: move to cpp

File: shm/Player.hpp [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+#include <memory>
+
+class Player {
+public:
+    Player(std::shared_ptr<Ship> s = nullptr, size_t m = 0, size_t as = 0)
+            :ship_(s), money_(m), availableSpace_(as) {}
```
Comment (nauka-programowania-MA) [inline shm/Player.hpp:11]: to .cpp

File: shm/Player.hpp [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+    std::shared_ptr<Ship> getShip() const { return ship_; }
+    size_t getMoney()               const { return money_; }
+    size_t getAvailableSpace()      const { return availableSpace_; }
```
Comment (nauka-programowania-MA) [inline shm/Player.hpp:15]: one style in whole code

**Review by nauka-programowania-MA — APPROVED** [review]

**Comment by nauka-programowania-MA** [comment]
{
"requirement_ids": [2568, 2568, 2568, 2568, 2568, 2568, 2568, 2568, 2569, 2570],
"points": [5, 5, 5, 5, 5, 5, 5, 5, 20, 6],
"users": ["wbrejecki", "Ian64", "McKucia", "kali-03", "DawidMalarz ", "malinaq3"]
}

File: shm/Player.cpp [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+void Player::countAvailableSpace() {
+
+    auto amoutOfCargo = ship_->getCargo().size();
+    auto availableSpaceTmp = ship_->getCapacity() - amoutOfCargo;
+
+    if(availableSpaceTmp <= 0)
+        availableSpace_ = 0;
+    else
+        availableSpace_ = availableSpaceTmp;
```
Comment (Kasperice) [inline shm/Player.cpp:20]: I see that you are using brackets in if/else, please use them here also

**Review by Kasperice — COMMENTED** [review]
I had only one comment (style). As nauka-programowania-MA said, it is worth to move constructor definition to cpp file, we also moved getters/setters to cpp file and now hpp is very clear (this is of course my opinion ;) )

**Pull request closed without merging**